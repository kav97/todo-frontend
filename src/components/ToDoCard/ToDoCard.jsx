import './ToDoCard.scss'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt, faPen } from "@fortawesome/free-solid-svg-icons";

const ToDoCard = ({ toDos = [], deleteToDo, noToDos}) => {
    if (toDos.length === 0) {
        return <div className="empty-list">{noToDos}</div>
    }

    return (
        <>
            {toDos.map((toDo) => {
                return (
                    <div className="todo-card" key={toDo.id}>
                        <p>{toDo.text}</p>
                        <p>{toDo.dateCreated}</p>
                        <button
                            className="list__edit"
                            aria-label="Edit todo item"
                            onClick={(e) => {
                                e.preventDefault();
                                window.location.href=`/update/${toDo.id}`;
                            }}
                        >
                        <FontAwesomeIcon icon={faPen} />
                        </button>
                        <button
                            className="list__delete"
                            aria-label="Delete todo item"
                            onClick={() => deleteToDo(toDo.id)}
                        >
                        <FontAwesomeIcon icon={faTrashAlt} />
                        </button>
                    </div>
                )
                
            })}
        </>
    )
}

export default ToDoCard