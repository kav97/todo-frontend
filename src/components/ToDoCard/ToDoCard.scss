@use "../../assets/styles/variables.scss" as *;

.todo-card {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: $color-blue;
    padding: 15px;
    border: solid 4px $color-black;
    border-radius: 10px;
    height: 250px;
    overflow-y: scroll;

    &__date{
        font-size: 1.25rem;
        font-style: italic;
    }

    &__buttons {
        display: flex;
        justify-content: space-between;

        &--update {
            padding:0.35rem;
            border-radius: 50%;
            background-color: $color-white;
            color: $color-black;
            border: none;
            cursor: pointer;
        }

        &--delete {
            padding:0.35rem;
            border-radius: 50%;
            background-color: $color-black;
            color: $color-white;
            border: none;
            cursor: pointer;
        }
    }
    
    &::-webkit-scrollbar {
        width: 5px;
        &-thumb {
            background: $color-yellow;
        }
    }
}