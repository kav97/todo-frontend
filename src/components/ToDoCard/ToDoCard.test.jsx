import { render } from "@testing-library/react";
import ToDoCard from "./ToDoCard"

it("Should render the each To Do Card to screen", () => {
  const { component } = render(<ToDoCard />);

  expect(component).toMatchSnapshot();
});