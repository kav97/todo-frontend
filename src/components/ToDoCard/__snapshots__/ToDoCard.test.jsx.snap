// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the each To Do Card to screen 1`] = `undefined`;
